wagtail/documents/views/bulk_actions/add_to_collection.py

```diff
@@ -0,0 +1,55 @@
+from django import forms
+from django.utils.translation import gettext_lazy as _
+from django.utils.translation import ngettext
+
+from wagtail.core import hooks
+from wagtail.core.models.collections import Collection
+from wagtail.documents.permissions import permission_policy
+from wagtail.documents.views.bulk_actions.document_bulk_action import DocumentBulkAction
+
+
+class CollectionForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['collection'] = forms.ModelChoiceField(
+            queryset=Collection.objects.all()
+        )
+
+
+class AddToCollectionBulkAction(DocumentBulkAction):
+    display_name = _("Add to collection")
+    action_type = "add_to_collection"
+    aria_label = _("Add documents to collection")
+    template_name = "wagtaildocs/bulk_actions/confirm_bulk_add_to_collection.html"
+    action_priority = 30
+
+    def check_perm(self, document):
+        return permission_policy.user_has_permission_for_instance(self.request.user, 'change', document)
```

Comment:
I think this permissions check is incomplete. The user also needs the `add` permission for the new collection.

Reply:
The `user_has_permission_for_instance` [here](https://github.com/wagtail/wagtail/blob/99f05d67cb97465127c095fc2664ff161e1b049a/wagtail/core/permission_policies/collections.py#L254), checks for the 'add' permission as well, if 'change' permission is checked for. Is it necessary to mention explicitly?

Reply:
Ah, bit of miscommunication here, I'm referring to the new collection. The user needs to have change permission on the existing instance in the old collection, but also needs to have permission to add a document to the new collection. (This doesn't necessarily have to be done here - it could be done by limiting the selectable collections in the form as I mention below - but it should be done somewhere)